lib/platform/types.ts

```diff
@@ -55,7 +55,6 @@ export interface Pr {
   displayNumber?: string;
   hasAssignees?: boolean;
   hasReviewers?: boolean;
-  isConflicted?: boolean;
```

Comment:
I think we should also use this as fast conflict check if available, so we don't need the manual merge test. That will speedup checks.

Reply:
Depends. For some platforms like Bitbucket it involves another API call specifically

Reply:
Also, I think @zharinov found in testing that GitHub even returns UNKNOWN instead of CONFLICTED many times, so it's not reliable (we know we can't merge it, but we may miss that it's conflicted and not rebase it)

Reply:
sure, but some platforms return conflicted as pr staste, so we don't need the git check. So i would prefer to leave the pr check when it's part of the fetched pr metadata and only use git check if pr is not already conflicted

Reply:
GitHub is not reliable (it can return UNKNOWN if it hasn't calculated CONFLICTED status yet), Bitbucket takes API call. Not sure it's worth then the special case of using PR.isConflicted if it's only for a subset. Also would be interested to learn how long this command takes to execute locally - @zharinov can you do some timing locally?

Reply:
BTW I suspect that the reason we sometimes see conflicted PRs not rebased when we expect to is because the GitHub backend is determining conflicted state asynchronously and returning UNKNOWN

Reply:
> From what I see, for non-conflicted branches it's about 100ms, while for the conflicted one it's about 1100ms

Reply:
UPD: same results for measuring started after `syncGit()` call

Reply:
> > can you do some timing locally?
> 
> From what I see, for non-conflicted branches it's about 100ms, while for the conflicted one it's about 1100ms

Can you further work out which of the steps takes up that time? e.g. is it mostly the attempted merge, or is it partly the branch checkouts and resets, etc?

Also I think we should aim to cache this information. e.g. record the baseBranchSha and branchSha and conflicted result and then reuse it in subsequent runs if unchanged - meaning only need to recompute it whenever the base branch is updated.

Reply:
Sure, I'll research it for more details

Reply:
> Can you further work out which of the steps takes up that time? e.g. is it mostly the attempted merge, or is it partly the branch checkouts and resets, etc?

As I can see, it's mostly because of syncing with remote. When we do checkout to `renovate/some-branch` for the first time manually, it shows me progress info on how remote objects are being received. And if I get it right, we're about to have this lag anyway, attempting to rebase branch. Other steps seem to be relatively instant (~20ms each).

But we have additional 500ms for each conflict check, and this is subject to my further work on caching.

Reply:
Yes, we shouldn't count the sync if we will need to do it anyway. I'm interested in the time to calculate the merge and reset, both in conflicted and non-conflicted state